File: artifacts/lodepng/translate/src/b_p_m_lists.rs
use crate::*;

#[derive(Clone)]
pub struct BPMLists {
    pub memsize: u32,
    pub memory: Option<Vec<BPMNode>>,
    pub numfree: u32,
    pub nextfree: u32,
    pub freelist: Option<Vec<Option<Box<BPMNode>>>>,
    pub listsize: u32,
    pub chains0: Option<Vec<Option<Box<BPMNode>>>>,
    pub chains1: Option<Vec<Option<Box<BPMNode>>>>,
}

File: artifacts/lodepng/decompose/src/b_p_m_lists.c
typedef struct BPMLists
{
  unsigned memsize;
  BPMNode *memory;
  unsigned numfree;
  unsigned nextfree;
  BPMNode **freelist;
  unsigned listsize;
  BPMNode **chains0;
  BPMNode **chains1;
} BPMLists
/*** DEPENDENCIES:
typedef struct BPMNode
{
  int weight;
  unsigned index;
  struct BPMNode *tail;
  int in_use;
} BPMNode
----------------------------
***/


